apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -305,15 +286,12 @@ export class CaseController {
   ) {
     const existingCase = await this.findCaseById(id)
 
-    if (user.role !== UserRole.JUDGE) {
-      throw new ForbiddenException('A ruling must be signed by a judge')
+    if (user.role !== UserRole.JUDGE && user.id !== existingCase.judge.id) {
```

Comment:
```suggestion
    if (user.role !== UserRole.JUDGE && user.id !== existingCase.judgeId) {
```

The database stores the judge id in the column judge_id which is available in the case model as the field judgeId. The judge fieled is filled in automatically by querying the database for a user by judgeId. It is safer to compare to judgeId, because it could be null and if so judge will also be null and we will get a null pointer reference exception.